mitodl/ccxcon
# Support for syncing modules.

#### What's this PR do?

Adds support for fetching course structure (aka Modules) from the edx API.
#### Where should the reviewer start?

courses/tasks.py
#### How should this be manually tested?
1. Create an [oauth application](http://localhost:8000/admin/oauth2/client/) in your edx devstack. You should create a backing user for the application as well. Type should be Confidential.
2. Create a [grant](http://localhost:8000/admin/oauth2/grant/) with the same backing user.
3. Take the tokens and username generated in 1 and 2 and add it to [ccxcon](https://localhost.daplie.com:8077/admin/oauth_mgmt/backinginstance/add/). The access token, refresh token, and expiration date should all stay as their default values (empty for tokens, 1970 for expiration).
4. Add user you created as a staff member of the course in question.
5. Load it up in `./manage.py shell` and trigger a save on the course you want to trigger an update for. Ensure it has a valid `course_id`.
6. `course.module_set.all()` should return a list of that course's modules in order.
#### Any background context you want to provide?

I've moved edx_instance to it's own fully-formed model so we can store additional OAuth information. This is in a new app called `oauth_mgmt`.

You can now exclude slow tests with `-m \"not slowtest\"` which will skip dredd from running.
#### What are the relevant tickets?

Fixes #91 #9 #5 
#### What gif best describes this PR or how it makes you feel?

![](https://media.giphy.com/media/OMeGDxdAsMPzW/giphy.gif)
